src/components/video-player/VideoPlayer.js

```diff
@@ -1,39 +1,113 @@
+import {Pressable} from 'react-native';
 import Video from 'react-native-video';
 import {VideoControls} from './video-controls';
+import {ControlBarTypes, ResponseOptionTypes} from './index';
 import styles from './styles';
 
 const VIDEO_PATH = require('../../../AssignmentVideo2.mp4');
 
-const VideoPlayer = () => {
+const VideoPlayer = ({navigation}) => {
+  const videoPlayerRef = useRef(null); // to take ref of VideoPlayer
   const [showControls, setShowControls] = useState(false);
+  const [showResponse, setShowResponse] = useState(false);
+  const [pause, setPause] = useState(false);
+  const [videoData, setVideoData] = useState({
+    totalDuration: null,
+    presentDuration: null,
+  });
+
   // handling errors
   const handleError = (error) => {
     console.log('Error while playing video: ', error);
   };
 
-  // listening to press events on video player
+  /**
+   * listening to press events of video player
+   * for showing/hiding player controls
+   */
   const handleVideoPress = () => {
     setShowControls(!showControls);
   };
 
+  // controlling video player to play/pause
+  const handlePlayPause = (value) => {
+    setPause(value);
+    setShowResponse(false);
+  };
+
+  // showing/hiding user responses
+  const handleResponseOptions = () => {
+    setShowResponse(!showResponse);
+  };
+
+  // storing video progress
+  const handleVideoProgress = ({currentTime, playableDuration}) => {
+    setVideoData({
+      totalDuration: playableDuration,
+      presentDuration: currentTime,
+    });
+  };
+
+  // handling seek - video position forward/backward
+  const handleSeek = (type) => {
+    switch (type) {
+      case ControlBarTypes.SEEK_FORWARD: {
+        videoPlayerRef.current.seek(videoData.presentDuration + 10);
+        return;
+      }
+      case ControlBarTypes.SEEK_BACKWARD: {
+        videoPlayerRef.current.seek(videoData.presentDuration - 10);
+        return;
+      }
+      default:
+        return;
+    }
+  };
+
+  /**
+   * handling user selected response option -
+   * by navigating to its respective screen
+   */
+  const handleUserSelectedResponseOption = (type) => {
+    switch (type) {
+      case ResponseOptionTypes.VIDEO_RESPONSE_OPTION: {
+        return;
+      }
+      case ResponseOptionTypes.AUDIO_RESPONSE_OPTION: {
+        return;
+      }
+      case ResponseOptionTypes.TEXT_RESPONSE_OPTION: {
+        return;
+      }
+      default:
+        return;
+    }
+  };
+
   return (
-    <TouchableOpacity style={styles.container} onPress={handleVideoPress}>
+    <Pressable style={styles.container} onPress={handleVideoPress}>
       <Video
-        paused={showControls}
+        ref={videoPlayerRef}
         source={VIDEO_PATH}
         resizeMode="stretch"
         onError={handleError}
         style={styles.backgroundVideo}
+        paused={pause}
+        onProgress={handleVideoProgress}
+      />
+      <VideoControls
+        showControls={showControls}
```

Comment:
Do you think we could have used context here? instead of passing these props across 2 hierarchy levels. First to Video controls and then to ResponseBar and ControlBar?

Reply:
Yes, this could be done more appropriately by using context api's. 
I'll update in next subsequent PR's.